service/src/main/java/com/zebrunner/reporting/service/reporting/ReportingService.java

```diff
@@ -0,0 +1,222 @@
+        oldTest.setTestCaseId(testCase.getId());
+        oldTest.setStartTime(Timestamp.valueOf(test.getStartedAt().atZoneSameInstant(ZoneOffset.UTC).toLocalDateTime()));
+
+        com.zebrunner.reporting.domain.db.Test existingTest = testMapper.getTestByTestRunIdAndUuid(testRunId, test.getUuid());
+        if (existingTest != null) {
+            oldTest.setId(existingTest.getId());
+        }
+
+        oldTest = testService.startTest(oldTest, null, null);
+
+        test.setId(oldTest.getId());
+        return test;
+    }
+
+    @Transactional
+    public Test finishTest(Test test, Long runId) {
+        Status status = Status.valueOf(test.getResult());
+
+        com.zebrunner.reporting.domain.db.Test oldTest = convertToOldTest(test, runId);
+        oldTest.setStatus(status);
+        oldTest.setMessage(test.getReason());
+        oldTest.setFinishTime(Timestamp.valueOf(test.getEndedAt().atZoneSameInstant(ZoneOffset.UTC).toLocalDateTime()));
+
+        testService.finishTest(oldTest, null, null);
+        return test;
+    }
+
+    @Transactional(readOnly = true)
+    public List<Test> getTestsByCiRunId(String ciRunId, List<Status> statuses, List<Long> testIds) {
+        List<com.zebrunner.reporting.domain.db.Test> oldTests = testService.getTestsByTestRunId(ciRunId);
+        return oldTests.stream()
+                       .filter(oldTest -> statuses == null || statuses.contains(oldTest.getStatus()))
+                       .filter(oldTest -> testIds == null || testIds.contains(oldTest.getId()))
+                       .map(this::convertToTest)
+                       .collect(Collectors.toList());
+    }
+
+    @Transactional(readOnly = true)
+    public com.zebrunner.reporting.domain.db.TestRun getTestRunFullById(Long id) {
+        return testRunService.getTestRunByIdFull(id);
+    }
+
+    @Transactional(readOnly = true)
+    public com.zebrunner.reporting.domain.db.Test getTestById(Long id) {
+        return testService.getTestById(id);
+    }
+
+    private TestSuite convertToSuite(TestRun testRun, Long userId) {
+        User user = new User();
+        user.setId(userId);
+
+        TestSuite testSuite = new TestSuite();
+        testSuite.setName(testRun.getName());
+        testSuite.setFileName(testRun.getName());
+        testSuite.setUser(user);
+
+        return testSuite;
+    }
+
+    private com.zebrunner.reporting.domain.db.TestRun convertToOldTestRun(TestRun testRun, String projectKey, Long userId) {
+        User user = new User();
+        user.setId(userId);
+
+        Project project = new Project();
+        project.setName(projectKey);
+
+        com.zebrunner.reporting.domain.db.TestRun oldTestRun = new com.zebrunner.reporting.domain.db.TestRun();
+        oldTestRun.setId(testRun.getId());
+        oldTestRun.setStartedBy(com.zebrunner.reporting.domain.db.TestRun.Initiator.HUMAN);
+        oldTestRun.setBuildNumber(1);
```

Comment:
Build number is always 1, is it updated later or not used anymore?

Reply:
yes